mlx/ops.cpp

```diff
@@ -1316,6 +1316,15 @@ array mean(
     const std::vector<int>& axes,
     bool keepdims /* = false */,
     StreamOrDevice s /* = {}*/) {
+  int ndim = a.ndim();
+  for (int axis : axes) {
+    if (axis < -ndim || axis >= ndim) {
+      throw std::invalid_argument(
+          "[mean] axis " + std::to_string(axis) +
+          " is out of bounds for array with " + std::to_string(ndim) +
+          " dimensions.");
+    }
+  }
```

Comment:
Could you use an `ostringstream` as we do elsewhere for consistency. See e.g. https://cautious-pancake-4gn87ze.pages.github.io/build/html/index.html

Otherwise looks great, thanks for adding this!

Reply:
it's done!